internal/s3fs/s3fs.go

```diff
@@ -63,6 +68,11 @@ func (f *FS) Open(name string) (fs.File, error) {
 		return nil, pathError
 	}
 
+	// If a prefix is provided, cannot Open files without the prefix
+	if f.prefix != "" && !strings.HasPrefix(name, f.prefix) {
+		return nil, pathError
+	}
```

Comment:
One thought on this:

I wonder if instead of enforcing that all keys provide the prefix, we should instead forcibly prepend the prefix.

My reasoning is for e.g. `.flipt.yml` if we dont do this, given a prefix is defined, the user will never be able to override the `.flipt.yml`. Since the storage layer will always request it without the prefix. This might be a confusing and easy mistaken to make.

However, if we forcibly apply the prefix, then the user can still put a `.flipt.yml` at the prefix in their bucket and get the same experience of controlling the overrides that way.

What do you think @ahobson ?

Reply:
Ahh yeah I think I see what you're saying there r.e. the subtle interaction there.
Kind of append if not already prefixed?:
```go
if !strings.HasPrefix(path, prefix) {
    path = prefix + path
}
```
That sounds great if what I am saying equates with what you're thinking 👍